integration_test/cases/repo.exs

```diff
@@ -106,6 +106,18 @@ defmodule Ecto.Integration.RepoTest do
            TestRepo.delete!(updated)
   end
 
+  @tag :focus
+  test "insert, update and delete with invalid field" do
+    post = %Post{title: "title"}
+    changeset =
+      post
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_change(:nonexistent_field, "foo")
```

Comment:
@josevalim Could you please advise what would be the desired behavior in this case?

Reply:
I can add some historical context: https://github.com/elixir-ecto/ecto/pull/1883#issuecomment-269858878. Back then, the consensus was that it's OK to store some additional information on the changeset.

At the same time though, I agree that non-existing fields shouldn't end up on the final struct so maybe we should filter them out before returning from repo operations.

Reply:
thanks, the PR is updated.